cgp/hl_api.py

```diff
@@ -69,11 +78,20 @@ def evolve(
             max_fitness = pop.champion.fitness
 
         if print_progress:
-            print(
-                f"\r[{pop.generation + 1}/{max_generations}] max fitness: {max_fitness}\033[K",
-                end="",
-                flush=True,
-            )
+            if not max_generations == np.inf:
+                print(
+                    f"\r[{pop.generation + 1}/{max_generations}] max fitness: {max_fitness}\033[K",
+                    end="",
+                    flush=True,
+                )
+            elif not max_evaluations == np.inf:
+                print(
+                    f"\r[{objective.calls}/{max_evaluations}] max fitness: {max_fitness}\033[K",
```

Comment:
doesn't this also need a `+1` like in the other if branch?

Reply:
I don't think so, since the `objective.calls` has no offset to the number of evaluations. What I just realized could happen though, is that there are too many `evaluations`. Since `ea.step` (https://github.com/HenrikMettler/hal-cgp/blob/212bdea9dd53207ba0f0ec097d2596d0bd2f9767/cgp/hl_api.py#L71) can call the `objective` function more than once (for multiple offspring with different phenotype). I don't know if there is an easy fix to this without moving the while loop for `objective.calls` into the `ea.step`